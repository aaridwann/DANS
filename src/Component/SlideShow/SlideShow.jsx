import React, { useState, useEffect } from "react";
import img_hero from "../../assets/img_hero.png";
import img_hero2 from "../../assets/img_hero2.png";
import img_hero3 from "../../assets/img_hero3.png";

const slides = [
  {
    id: 1,
    image: img_hero,
    caption: "Slide 1",
  },
  {
    id: 2,
    image: img_hero2,
    caption: "Slide 2",
  },
  {
    id: 3,
    image: img_hero3,
    caption: "Slide 3",
  },
];

function SlideShow() {
  const [currentSlide, setCurrentSlide] = useState(0);

  const goToPrevSlide = () => {
    setCurrentSlide((currentSlide - 1 + slides.length) % slides.length);
  };

  const goToNextSlide = () => {
    setCurrentSlide((currentSlide + 1) % slides.length);
  };

  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentSlide((currentSlide + 1) % slides.length);
    }, 3000);
    return () => clearInterval(interval);
  }, [currentSlide]);

  return (
    <div className="relative h-full">
      {slides.map((slide, index) => (
        <div
          key={slide.id}
          className={`absolute top-0 left-0 w-full h-full transition-opacity duration-500 ${
            index === currentSlide ? "opacity-100" : "opacity-0"
          }`}
        >
          <img
            src={slide.image}
            alt={slide.caption}
            className="w-auto drop-shadow-4xl mx-auto h-full object-cover"
          />
        </div>
      ))}
    </div>
  );
}
export default SlideShow;
