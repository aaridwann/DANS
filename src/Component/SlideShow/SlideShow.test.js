import React from "react";
import { render } from "@testing-library/react";
import SlideShow from "./SlideShow";
import "@testing-library/jest-dom/extend-expect";

describe("SlideShow component", () => {
  test("renders SlideShow component", () => {
    const { getByAltText } = render(<SlideShow />);
    const slide1 = getByAltText("Slide 1");
    expect(slide1).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<SlideShow />);
    expect(container.firstChild).toMatchSnapshot();
  });
});