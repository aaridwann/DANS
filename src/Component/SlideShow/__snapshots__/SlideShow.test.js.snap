// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideShow component matches snapshot 1`] = `
<div
  class="relative h-full"
>
  <div
    class="absolute top-0 left-0 w-full h-full transition-opacity duration-500 opacity-100"
  >
    <img
      alt="Slide 1"
      class="w-auto drop-shadow-4xl mx-auto h-full object-cover"
      src="mocked-image"
    />
  </div>
  <div
    class="absolute top-0 left-0 w-full h-full transition-opacity duration-500 opacity-0"
  >
    <img
      alt="Slide 2"
      class="w-auto drop-shadow-4xl mx-auto h-full object-cover"
      src="mocked-image"
    />
  </div>
  <div
    class="absolute top-0 left-0 w-full h-full transition-opacity duration-500 opacity-0"
  >
    <img
      alt="Slide 3"
      class="w-auto drop-shadow-4xl mx-auto h-full object-cover"
      src="mocked-image"
    />
  </div>
</div>
`;
